Compiler/src/ssair/inlining.jl

```diff
@@ -975,6 +975,14 @@ function retrieve_ir_for_inlining(mi::MethodInstance, ir::IRCode, preserve_local
     ir.debuginfo.def = mi
     return ir, spec_info, DebugInfo(ir.debuginfo, length(ir.stmts))
 end
+function retrieve_ir_for_inlining(mi::MethodInstance, opt::OptimizationState, preserve_local_sources::Bool)
+    result = opt.result
+    if result !== nothing
+        !result.simplified && simplify_ir!(result)
+        return retrieve_ir_for_inlining(mi, result.ir, preserve_local_sources)
+    end
+    retrieve_ir_for_inlining(mi, opt.src, preserve_local_sources)
+end
```

Comment:
Is this new `retrieve_ir_for_inlining` (and the new `OptimizationState` handler of `src_inlining_policy`) tested somewhere? IIUC any test that would hit this.

Reply:
Never mind, because `transform_result_for_cache` in `finish!` is only applied to `result` that will be cached (due to `isdefined(result, :ci)`), inference local results and will keep the `OptimizationState` as is. So these new inlining code should be actually hit during the native compilation.